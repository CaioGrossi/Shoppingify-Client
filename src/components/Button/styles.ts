import styled, { css } from 'styled-components';

import { ButtonProps } from '.';

const wrapperModifiers = {
  minimal: () => css`
    background: none;
    color: black;
  `,
  fullWidth: () => css`
    width: 100%;
  `
};

type WrapperProps = Pick<ButtonProps, 'minimal' | 'fullWidth'>;

export const Wrapper = styled.button<WrapperProps>`
  ${({ minimal, fullWidth }) => css`
    cursor: pointer;
    font-weight: 600;
    background-color: #f9a109;
    color: white;
    font-size: 2rem;
    border: none;
    border-radius: 12px;
    width: 8.7rem;
    display: flex;
    justify-content: center;
    align-items: center;
    height: 5rem;

    ${minimal && wrapperModifiers.minimal()}
    ${fullWidth && wrapperModifiers.fullWidth()};
  `}
`;
