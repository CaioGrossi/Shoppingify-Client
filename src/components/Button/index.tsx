import { ButtonHTMLAttributes } from 'react';
import * as S from './styles';

export type ButtonProps = {
  fullWidth?: boolean;
  minimal?: boolean;
  children: React.ReactNode;
} & ButtonHTMLAttributes<HTMLButtonElement>;

const Button = ({
  children,
  fullWidth = false,
  minimal = false,
  ...props
}: ButtonProps) => (
  <S.Wrapper fullWidth={fullWidth} minimal={minimal} {...props}>
    <span>{children}</span>
  </S.Wrapper>
);

export default Button;
