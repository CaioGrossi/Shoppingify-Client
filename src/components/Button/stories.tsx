import { Story, Meta } from '@storybook/react/types-6-0';
import Button, { ButtonProps } from '.';

export default {
  title: 'Button',
  component: Button
} as Meta;

export const Default: Story<ButtonProps> = () => <Button>Save</Button>;

export const Minimal: Story<ButtonProps> = () => (
  <Button minimal={true}>Minimal</Button>
);
