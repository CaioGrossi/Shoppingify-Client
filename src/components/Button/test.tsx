import { render, screen } from '@testing-library/react';

import Button from '.';

describe('<Button />', () => {
  it('should render correctly', () => {
    const { container } = render(<Button>save</Button>);

    expect(screen.getByRole('button', { name: /save/i })).toBeInTheDocument();

    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render with 100% size when fullWidth is passed', () => {
    render(<Button fullWidth>save</Button>);

    expect(screen.getByRole('button', { name: /save/i })).toHaveStyleRule(
      'width',
      '100%'
    );
  });

  it('should render with none background when minimal is passed', () => {
    render(<Button minimal>save</Button>);

    expect(screen.getByRole('button', { name: /save/i })).toHaveStyleRule(
      'background',
      'none'
    );
  });
});
