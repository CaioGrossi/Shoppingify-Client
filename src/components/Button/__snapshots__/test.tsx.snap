// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> should render correctly 1`] = `
.c0 {
  cursor: pointer;
  font-weight: 600;
  background-color: #f9a109;
  color: white;
  font-size: 2rem;
  border: none;
  border-radius: 12px;
  width: 8.7rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 5rem;
}

<button
  class="c0"
>
  <span>
    save
  </span>
</button>
`;
